### PR TITLE
feat: consolidated yolov5s effnetb0

### PR DESCRIPTION
**Pull request by vli11**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
adding consolidated yolov5s eenetb0 script

## Issue this PR will close

close: #**issue_number**

## Test Instructions if applicable
test the script inside the container

### CODE REVIEW

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	exit 0
+fi
+
+# arc only
+# GST_VAAPI_DRM_DEVICE=/dev/dri/renderD129
+# export GST_VAAPI_DRM_DEVICE="$GST_VAAPI_DRM_DEVICE"
+
+DEVICE="CPU" #GPU|CPU|MULTI:GPU,CPU
+PRE_PROCESS="pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1 batch-size=1 nireq=4 gpu-throughput-streams=4" #""|pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1 batch-size=1 nireq=4 gpu-throughput-streams=4 
+PRE_PROCESS2="pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1 reclassify-interval=1 batch-size=1 nireq=4 gpu-throughput-streams=4" #""|pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1 reclassify-interval=1 batch-size=1 nireq=4 gpu-throughput-streams=4
```
Comment (jim-wang-yutsung) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh]: what are pre-process and pre_process2?  please use more descriptive name to indicate the cases

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Copyright (C) 2023 Intel Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if [ "$INPUTSRC_TYPE" == "REALSENSE" ]; then
+        decode_pp="! videoconvert ! video/x-raw,format=BGR"
+	# TODO: update with vaapipostproc when MJPEG codec is supported.
+	echo "Not supported until D436 with MJPEG." > /tmp/results/pipeline$cid_count.log
+	exit 0
```
Comment (jim-wang-yutsung) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh]: should be exit 2 if not supported

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+fi
+
+# arc only
+# GST_VAAPI_DRM_DEVICE=/dev/dri/renderD129
+# export GST_VAAPI_DRM_DEVICE="$GST_VAAPI_DRM_DEVICE"
```
Comment (jim-wang-yutsung) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh]: should we remove this?

**Review by jim-wang-yutsung — CHANGES_REQUESTED** [review]
i don't understand the context of this PR but giving some comments based on what i have anyway

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+if [ "$INPUTSRC_TYPE" == "REALSENSE" ]; then
+        decode_pp="! videoconvert ! video/x-raw,format=BGR"
+	# TODO: update with vaapipostproc when MJPEG codec is supported.
+	echo "Not supported until D436 with MJPEG." > /tmp/results/pipeline$cid_count.log
+	exit 2
+fi
+
+DECODE="vaapidecodebin"
+DEVICE="CPU" #GPU|CPU|MULTI:GPU,CPU
+PRE_PROCESS="" #""|pre-process-backend=vaapi-surface-sharing|pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1
```
Comment (jim-wang-yutsung) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:17]: should default to the suitable value; asking the end user to put all this thing is too difficult to use in terms of usability or poor experience.

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            # error 'ERROR: "--device" requires an argument GPU|CPU|MULTI:GPU,CPU.'
+        fi
+        ;;
+    --pre_process)
+        if [ "$2" ]; then
+            PRE_PROCESS=$2
+            shift
+        else
+            echo "using default"
+            # error 'ERROR: "--pre_process" requires an argument ""|pre-process-backend=vaapi-surface-sharing|pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1 .'
```
Comment (jim-wang-yutsung) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:52]: in my opinion, we shouldn't ask user to do this

**Review by jim-wang-yutsung — CHANGES_REQUESTED** [review]

**Review by jim-wang-yutsung — CHANGES_REQUESTED** [review]
please address my comments thanks

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh (hunk 4 above)
Comment (vli11) [reply configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:17]: the value here is more for advanced user to play with; I believe this is primitive of what we will do for user, right now this is only to consolidate the codes in one place, @brian-intel has plan to do more to this

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh (hunk 5 above)
Comment (vli11) [reply configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:52]: this is primitive of what we will do for user, right now this is only to consolidate the codes in one place, @brian-intel has plan to do more to this for user

Comment (jim-wang-yutsung) [reply configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:52]: could we remove this flag and just use if-else logic to take care of that for now?

File: configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	# TODO: update with vaapipostproc when MJPEG codec is supported.
+	echo "Not supported until D436 with MJPEG." > /tmp/results/pipeline$cid_count.log
+	exit 2
+fi
+
+DECODE="vaapidecodebin"
+DEVICE="CPU" #GPU|CPU|MULTI:GPU,CPU
+PRE_PROCESS="" #""|pre-process-backend=vaapi-surface-sharing|pre-process-backend=vaapi-surface-sharing pre-process-config=VAAPI_FAST_SCALE_LOAD_FACTOR=1
+
+while :; do
```
Comment (Brian-McGinn) [inline configs/dlstreamer/framework-pipelines/yolov5_pipeline/yolov5s_effnetb0.sh:19]: Will be good once flag loop is removed

**Comment by vli11** [comment]
@jim-wang-intel as we have discussed this morning regarding your comments, the plan is to merge this PR into @brian-intel 's consolidateDLStreamer, then we will work on the ENV to replace the current parameter. with that discussion, can I resolve your comments here?

**Pull request merged**